src/extension.ts

```diff
@@ -111,12 +111,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
   if (lsPath.hasCustomBinPath()) {
     reporter.sendTelemetryEvent('usePathToBinary');
   }
-  const executable = await getServerExecutable(lsPath);
-  const serverOptions: ServerOptions = {
-    run: executable,
-    debug: executable,
-  };
-  outputChannel.appendLine(`Launching language server: ${executable.command} ${executable.args?.join(' ')}`);
+  const serverOptions: ServerOptions = await getServerOptions(lsPath);
```

Comment:
```suggestion
  const serverOptions = await getServerOptions(lsPath);
```

TS can infer the return type from `getServerOptions` here